flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/DeleteOrphanFiles.java

```diff
@@ -0,0 +1,313 @@
+              .forceNonParallel();
+
+      SingleOutputStreamOperator<String> filesToDelete =
+          tableMetadataFiles
+              .union(tableDataFiles)
+              .map(new FileUriConverter(equalSchemes, equalAuthorities))
+              .keyBy(FileURI::path)
+              .connect(
+                  allFsFiles
+                      .map(new FileUriConverter(equalSchemes, equalAuthorities))
+                      .keyBy(FileURI::path))
+              .process(new AntiJoin(prefixMismatchMode))
+              .slotSharingGroup(slotSharingGroup())
+              .name(operatorName(FILTER_FILES_TASK_NAME))
+              .uid(FILTER_FILES_TASK_NAME + uidSuffix())
+              .setParallelism(parallelism());
+
+      DataStream<Exception> errorStream =
+          tableMetadataFiles
+              .getSideOutput(ERROR_STREAM)
+              .union(
+                  allFsFiles.getSideOutput(ERROR_STREAM),
+                  tableDataFiles.getSideOutput(ERROR_STREAM),
+                  splits.getSideOutput(ERROR_STREAM),
+                  filesToDelete.getSideOutput(ERROR_STREAM));
+
+      // Stop deleting the files if there is an error
+      SingleOutputStreamOperator<String> filesOrSkip =
+          filesToDelete
+              .connect(errorStream)
+              .transform(
+                  operatorName(SKIP_ON_ERROR_TASK_NAME),
+                  TypeInformation.of(String.class),
+                  new SkipOnError())
+              .uid(SKIP_ON_ERROR_TASK_NAME + uidSuffix())
+              .slotSharingGroup(slotSharingGroup())
+              .forceNonParallel();
+
+      // delete the files
+      filesOrSkip
+          .rebalance()
+          .transform(
+              operatorName(DELETE_FILES_TASK_NAME),
+              TypeInformation.of(Void.class),
+              new DeleteFilesProcessor(
+                  tableLoader().loadTable(), taskName(), index(), deleteBatchSize))
+          .uid(DELETE_FILES_TASK_NAME + uidSuffix())
+          .slotSharingGroup(slotSharingGroup())
+          .setParallelism(parallelism());
+
+      // Ignore the file deletion result and return the DataStream<TaskResult> directly
+      return trigger
+          .connect(errorStream)
+          .transform(
+              operatorName(AGGREGATOR_TASK_NAME),
+              TypeInformation.of(TaskResult.class),
+              new TaskResultAggregator(tableName(), taskName(), index()))
+          .uid(AGGREGATOR_TASK_NAME + uidSuffix())
+          .slotSharingGroup(slotSharingGroup())
+          .forceNonParallel();
+    }
+  }
+
+  private static Map<String, String> flattenMap(Map<String, String> map) {
+    Map<String, String> flattenedMap = Maps.newHashMap();
+    if (map != null) {
+      for (String key : map.keySet()) {
+        String value = map.get(key);
+        for (String splitKey : COMMA_SPLITTER.split(key)) {
+          flattenedMap.put(splitKey.trim(), value.trim());
+        }
+      }
+    }
```

Comment:
nit: newline - not important at this point, just marking so we can fix if it is still relevant after the reviews